packages/devtools/src/commands/isElementDisplayed.js

```diff
@@ -1,11 +1,24 @@
+import executeScript from './executeScript'
+
 import { getStaleElementError } from '../utils'
+import { ELEMENT_KEY } from '../constants'
+
+const SCROLL_INTO_VIEW_SCRIPT = 'return (function scrollIntoView (elem) { elem.scrollIntoView() }).apply(null, arguments)'
 
-export default function isElementDisplayed ({ elementId }) {
+export default async function isElementDisplayed ({ elementId }) {
     const elementHandle = this.elementStore.get(elementId)
 
     if (!elementHandle) {
         throw getStaleElementError(elementId)
     }
 
+    /**
+     * scroll element into view
+     */
+    await executeScript.call(this, {
```

Comment:
I don't really like the idea to use scrollIntoView in isDisplayed (I'll provide an example why in next comment).
Are there any alternatives to it?
I assumed that isDisplayed should work even if element is not in current viewport.

Reply:
Open https://go.moplay.com/de/sports/registration-pop-up/?Offer=Bet10Get30GenSports
wait for site to load and form to open, then do in browser
`$$('[data-step="1"] .stepwise-registration__ssl')[0].scrollIntoView()`
That's it, form is broken :(

I use either `scrollIntoViewIfNeeded(false)` or `scrollIntoView(false)` here.
Maybe we can at least change `elem.scrollIntoView()` to `elem.scrollIntoViewIfNeeded()` ?

Reply:
So for the protocol command it doesn't matter if the element is within the viewport or not for determine the visibility. Since Puppeteer only allows to check if an element is in the viewport, I have to do this action before calling the isVisible command.